infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/KubernetesInternalRuntime.java

```diff
@@ -266,6 +271,58 @@ protected void internalStart(Map<String, String> startOptions) throws Infrastruc
     }
   }
 
+  private void fixCommands(KubernetesRuntimeContext<E> context) {
+    context
+        .getEnvironment()
+        .getCommands()
+        .stream()
+        .filter(c -> c.getPreviewUrl() != null)
+        .forEach(
+            command -> {
+              String machineName = command.getAttributes().get(MACHINE_NAME_ATTRIBUTE);
+              Optional<Ingress> ingress =
+                  context
+                      .getEnvironment()
+                      .getIngresses()
+                      .values()
+                      .stream()
+                      .filter(
+                          s -> {
+                            Map<String, String> annotations = s.getMetadata().getAnnotations();
+                            boolean hasMachineName =
+                                annotations.get("org.eclipse.che.machine.name").equals(machineName);
+                            if (hasMachineName) {
+                              List<String> anonPortKeys =
+                                  annotations
+                                      .keySet()
+                                      .stream()
+                                      .filter(k -> k.contains("port"))
+                                      .collect(Collectors.toList());
+                              for (String k : anonPortKeys) {
+                                if (annotations
+                                    .get(k)
+                                    .contains("" + command.getPreviewUrl().getPort() + "/")) {
+                                  return true;
+                                }
+                              }
+                            }
+                            return false;
+                          })
+                      .findFirst();
+              if (ingress.isPresent()) {
+                LOG.info("found ingress for previewurl [{}]", ingress.get());
+                String previewUrl =
+                    "http://" + ingress.get().getSpec().getRules().get(0).getHost() + "/";
+                if (command.getPreviewUrl().getPath() != null) {
+                  previewUrl += command.getPreviewUrl().getPath();
+                }
+                command.getAttributes().put(PREVIEW_URL_ATTRIBUTE, previewUrl);
+              } else {
+                LOG.info("ingress not found for [{}]", command);
```

Comment:
The corresponding warning should be propagated to k8s context instead of printing a logs.

Reply:
resolved. thanks for help with this